File: desafio-04/TeijiW/python/desafio04.py
"""
Author > Teiji Watanabe
Obtain the quantity of each piece without using a conditional or multiple choice structure
"""

PIECES = [
    [4, 3, 2, 5, 6, 2, 3, 4],
    [1, 1, 1, 1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1, 1, 1, 1],
    [4, 3, 2, 5, 6, 2, 3, 4],
]

PIECES_NAMES = {
    "Peão": 1,
    "Bispo": 2,
    "Cavalo": 3,
    "Torre": 4,
    "Rainha": 5,
    "Rei": 6
}


def main():
    """Main Function"""
    pieces_count = {
        0: 0,
        1: 0,
        2: 0,
        3: 0,
        4: 0,
        5: 0,
        6: 0,
    }
    for i, _ in enumerate(PIECES):
        for j in range(0, 7):
            pieces_count[j] += PIECES[i].count(j)

    for key in PIECES_NAMES:
        print("{}: {} peça(s)".format(key, pieces_count[PIECES_NAMES[key]]))


if __name__ == "__main__":
    main()

File: desafio-04/ThiagoCostaBarbosa/javascript/desafio04.js
function Chess_count(array) {
  let pieces = [0, "Peão", "Bispo", "Cavalo", "Torre", "Rainha", "Rei"],
      values = [0, 0, 0, 0, 0, 0, 0];

  //Remove Subarray
  array = array.flat();

  //Count number of pieces
  for (let i in array) {
    values[array[i]] += 1;
  }

  //Print the quantity of every piece
  console.log('\nQuantidade de peças:');
  for (i = 1; i < values.length; i++) {
    console.log(`${pieces[i]}: ${values[i]} peça(s)`);
  }
};

let tabuleiro = [
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 1, 1, 0, 0, 0],
  [0, 0, 0, 1, 1, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0]]

let tabuleiro2 = [
  [4, 3, 2, 5, 6, 2, 3, 4],
  [1, 1, 1, 1, 1, 1, 1, 1],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0],
  [1, 1, 1, 1, 1, 1, 1, 1],
  [4, 3, 2, 5, 6, 2, 3, 4]]

Chess_count(tabuleiro);
Chess_count(tabuleiro2);